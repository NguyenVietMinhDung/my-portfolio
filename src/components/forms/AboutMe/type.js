import type { Props as Skills } from '../../commons/ProgressBars/type';
import type { Props as BasicInfo } from './components/BasicInformation/type';

export type Props = {
  title: string,
  greeting: string,
  summary: string,
  basicInfo: BasicInfo,
  skills: Skills,
};
