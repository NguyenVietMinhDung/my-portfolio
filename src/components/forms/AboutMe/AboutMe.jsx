// @flow
import React, { memo } from 'react';
import CSSModules from 'react-css-modules';
import styles from './AboutMe.scss';
import type { Props } from './type';
import BasicInformation from './components/BasicInformation';
import { ProgressBars } from '../../commons';

const AboutMe = (props: Props) => {
  const {
    title, greeting, summary, basicInfo, skills,
  } = props;
  return (
    <div
      id="about-me"
      styleName="background"
    >
      <div
        styleName="container"
        data-aos="fade-right"
        data-aos-duration="2000"
      >
        <div styleName="title">{title}</div>
        <div styleName="content">
          <div styleName="introduction">
            <div styleName="greeting">{greeting}</div>
            <div>{summary}</div>
            <BasicInformation items={basicInfo} />
          </div>
          <div styleName="skills">
            <ProgressBars items={skills} />
          </div>
        </div>
      </div>
    </div>
  );
};

export default memo<Props>(CSSModules(AboutMe, styles));
