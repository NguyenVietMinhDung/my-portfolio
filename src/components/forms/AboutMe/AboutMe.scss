@import '../../../styles/abstracts/variables';
@import '../../../styles/abstracts/mixins';

@include form($white-color);

.content {
	display: flex;
	flex-flow: row wrap;
	justify-content: space-between;

	.introduction {
		display: flex;

		padding: 25px 0;
		align-items: flex-start;
		flex-flow: column wrap;

		.greeting {
			@include font(inherit, inherit, $black-color, 18px, 500);
			padding-bottom: 15px;
		}

		@media (min-width: $layout-breakpoint) {
			flex: 0.45;
		}

		@media (max-width: $layout-breakpoint) {
			flex: auto;
		}
	}

	.skills {
		display: flex;

		padding: 25px 0;
		align-items: flex-end;
		flex-flow: column wrap;

		@media (min-width: $layout-breakpoint) {
			flex: 0.45;
		}

		@media (max-width: $layout-breakpoint) {
			flex: auto;
		}
	}
}
