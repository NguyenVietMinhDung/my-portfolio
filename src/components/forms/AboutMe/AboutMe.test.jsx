import React from 'react';
import { shallow } from 'enzyme';
import {
  TITLE, GREETING, SUMMARY, basicInfo, skills,
} from '../../../constants/aboutMe';
import AboutMe from './AboutMe';

const defaultProps = {
  title: TITLE,
  greeting: GREETING,
  summary: SUMMARY,
  basicInfo,
  skills,
};

describe('<AboutMe />', () => {
  test('renders correctly', () => {
    const component = shallow(<AboutMe {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
