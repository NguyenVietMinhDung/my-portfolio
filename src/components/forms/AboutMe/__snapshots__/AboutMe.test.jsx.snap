// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutMe /> renders correctly 1`] = `
<div
  className="background"
  id="about-me"
>
  <div
    className="container"
    data-aos="fade-right"
    data-aos-duration="2000"
  >
    <div
      className="title"
    >
      About Me
    </div>
    <div
      className="content"
    >
      <div
        className="introduction"
      >
        <div
          className="greeting"
        >
          Hi There! I'm Dung Nguyen
        </div>
        <div>
          I am an ambitious Web
Application Developer who
always tries to improve himself
everyday, a quick learner who
can absorb new ideas and can
communicate clearly and
effectively. Looking for an
ambitious company which will
challenge my developer and
allows me to continue
to develop my knowledge and
potential.
        </div>
        <Memo(BasicInformation)
          items={
            Array [
              Object {
                "iconClassName": "lni-calendar",
                "name": "Date of birth",
                "value": "16 January 1995",
              },
              Object {
                "iconClassName": "lni-envelope",
                "name": "Email",
                "value": "ngvietminhdung@gmail.com",
              },
              Object {
                "iconClassName": "lni-phone-handset",
                "name": "Phone",
                "value": "(076) 401-2912",
              },
              Object {
                "iconClassName": "lni-map-marker",
                "name": "Location",
                "value": "Tan Binh District, HCM City",
              },
            ]
          }
        />
      </div>
      <div
        className="skills"
      >
        <Memo(ProgressBars)
          items={
            Array [
              Object {
                "itemId": "skill-1",
                "name": "JavaScript, HTML, CSS",
                "value": 80,
              },
              Object {
                "itemId": "skill-2",
                "name": "ReactJS",
                "value": 60,
              },
              Object {
                "itemId": "skill-3",
                "name": "SCSS",
                "value": 60,
              },
              Object {
                "itemId": "skill-5",
                "name": "Webpack",
                "value": 50,
              },
              Object {
                "itemId": "skill-6",
                "name": "Node.js",
                "value": 60,
              },
              Object {
                "itemId": "skill-7",
                "name": "Git",
                "value": 60,
              },
            ]
          }
        />
      </div>
    </div>
  </div>
</div>
`;
